app/views/teachers/TeacherCourseSolutionView.coffee

```diff
@@ -33,20 +33,30 @@ module.exports = class TeacherCourseSolutionView extends RootView
       return '' if l isnt @language
       a
 
+  getText: (thang, what) ->
+    if thang.i18n
+      if i18n.lng() in Object.keys(thang.i18n)
+        return thang.i18n[i18n.lng()][what]
+    return thang[what]
+
   onLoaded: ->
+    @listenTo me, 'change:preferredLanguage', @refreshData
+    @refreshData()
+
+  refreshData: ->
```

Comment:
Wish there were a better name than `refreshData`, but that function does do a lot of things so it's hard to have a good name for it. "updateLevelData" and pull out the lines that create `levelNumberMap` and render the page, I think.